client/client_status.go

```diff
@@ -81,13 +81,13 @@ func (u *StatusClient) Report(api ApiRequester, url string, report StatusReport)
 	switch {
 	case r.StatusCode == http.StatusConflict:
 		log.Warnf("status report rejected, deployment aborted at the backend")
-		return ErrDeploymentAborted
+		return NewAPIError(ErrDeploymentAborted, r)
 	case r.StatusCode != http.StatusNoContent:
 		log.Errorf("got unexpected HTTP status when reporting status: %v", r.StatusCode)
-		return errors.Errorf("reporting status failed, bad status %v", r.StatusCode)
+		return NewAPIError(errors.Errorf("reporting status failed, bad status %v", r.StatusCode), r)
 	}
 
-	log.Debugf("status reported, response %v", r)
+	log.Debugf("(request_id: %s): status reported, response %s", r.Header.Get("request_id"), r)
```

Comment:
Not sure if this will ever be the case, but do you know if we can end up with the empty `Header`? If so this might core dump if we are not checking `if Header != nil`.

Reply:
according to @maciejmrowiec the header is a part of every request. But maybe we should remove this, just in case it changes sometime in the future?

Reply:
I settled on removing them.